spack.yaml

```diff
@@ -3,40 +3,66 @@
 # It describes a set of packages to be installed, along with
 # configuration settings.
 spack:
-  definitions:
-  - compilers:
-    - intel@2021.10.0
-  - mpis:
-    - openmpi@4.1.5
-  - serial_packages:
-    - fortranxml@4.1.2
-  - packages:
-    - esmf@8.5.0+debug
-    - parallelio@2.6.2 build_type=RelWithDebInfo fflags='-qno-opt-dynamic-align -convert
-      big_endian -assume byterecl -ftz -traceback -assume realloc_lhs -fp-model source'
-      cflags='-qno-opt-dynamic-align -fp-model precise -std=gnu99'
-    - netcdf-c@4.9.2 build_system=cmake build_type=RelWithDebInfo
-    - netcdf-fortran@4.6.1
-    - fms@2023.02 precision=64 +large_file +deprecated_io ~gfs_phys ~openmp ~quad_precision
-      build_type=RelWithDebInfo
-    - access-om3 +install_libraries
   specs:
-  - matrix:
-    - [$mpis]
-    - [$%compilers]
-  - matrix:
-    - [$packages]
-    - [$^mpis]
-    - [$%compilers]
-  - matrix:
-    - [$serial_packages]
-    - [$%compilers]
+    - access-om3@git.2024.09.0
+  packages:
+    # Main Dependencies
+    access-om3-nuopc:
+      require:
+        - '@git.0.3.1'
+
+    # Other Dependencies
+    esmf:
+      require:
+        - '@8.5.0'
+    parallelio:
+      require:
+        - '@2.6.2'
+    netcdf-c:
+      require:
+        - '@4.9.2'
+        - build_system=cmake build_type=RelWithDebInfo
+    netcdf-fortran:
+      require:
+        - '@4.6.1'
+    fms:
+      require:
+        - '@2023.02'
+    openmpi:
+      require:
+        - '@4.1.5'
+    fortranxml:
+      require:
+        - '@4.1.2'
+
+    all:
+      require:
+        - '%intel@2021.10.0'
+        - target=x86_64
```

Comment:
Probably should put quotes around `target=x86_64`

Reply:
Done in https://github.com/ACCESS-NRI/ACCESS-OM3/pull/5/commits/0672015917f7b42b4b5a0717d5281f81dded8df6